### DISCUSSION TITLE
[ARK: Survival Ascended] Add completed boss fights to GSA dashboard character profile (if possible)

### DISCUSSION THREAD

**Discussion by DieselEffect** [gameserverapp/Platform#816] [Suggestions]
When a player completes a boss fight, the completed fight is added to their "Specimen Implant" in-game. Is it possible to get this information into the GSA dashboard?

This would be extremely helpful for recovering a character for someone who's character disappeared into the abyss or ended up detached from their account (happens all the time). For context, we currently have to ask players to post a screenshot of their in-game implant (as seen below) into a discord channel every time they finish a boss fight to track this information.

Example information:
![image](https://github.com/gameserverapp/Platform/assets/34605635/b9605335-91f6-4933-bfc5-08e643d0f309)

**Comment by altairhalo** [comment]
You can see some information in a `.arkprofile`.  However I am not 100% sure how GSA pulls the info for the dashboard,
`PerMapNamedExplorerNoteUnlocks`
`EmoteUnlocks`
`PlayerState_EngramBlueprints`
`NumChibiLevelUpsData`
`AscensionData (Boss Fights)`

**Reply by InfinityofphoenixTimo** [reply]
Adding Informations how Explorer Notes and the completed Bossfights to the dashboard, would be so awesome!

**Comment by mvaessen** [comment]
since this is very specific to ark, is this not something I see any benefits.
I'll leave it open for now, since it might be suitable for achievements.

**Reply by DieselEffect** [reply]
It would absolutely be beneficial for achievements. Boss defeated, or a collection of bosses defeated could result in a special role or reward such as tokens. I feel like this would only draw more people to our servers.

**Reply by mvaessen** [reply]
In theory this should already be accessible, by using the webhook api. Only thing missing is a mod that triggers said webhooks on certain in-game events. I'm not a mod dev, so I'm uncertain whether the devkit can plugin into those events. But I assume it does.

https://docs.gameserverapp.com/dashboard/automate_tasks/getting_started#webhook

https://docs.gameserverapp.com/developers/system-api/execute-task

**Reply by Buck-Irl** [reply]
So I'm looking into this. 
I feel logging bossfights would be really handy.

So two questions for you Max.

First. Can a webhook trigger achievement? I don't see option, now maybe I missed it. Custom roles is the only way I see. But that would be alot of roles🤔 

Second. Each player would need a webhook sent. 
I'm not sure how the api is set up, whether it's restapi but as dalo pointed out, if you send 2 webhooks in quick succession , it doesn't catch the second one.
So would a queue be possible? Or just stagger sending of the webhooks?

**Reply by mvaessen** [reply]
Webhooks cannot grant achievements atm, but that should not be hard or time consuming to add.

Rate limits or throttling of requests usually depends on whether you're calling the same webhook or not.
Cannot think of any limitations there atm, but it might depend on how things were set up.
